### PR TITLE
feat(api,worker): add minimum coverage threshold (#3502)

### PR DESCRIPTION
**Pull request by ziirish**
1. Description

Allow to setup a minimal coverage threshold

2. Related issues

#3502 

3. About tests
4. Mentions

@ovh/cds

Signed-off-by: Benjamin SANS <github@ziirish.info>

### CODE REVIEW

File: engine/worker/builtin_coverage.go [hunk 1]
```diff
@@ -70,6 +85,18 @@ func runParseCoverageResultAction(w *currentWorker) BuiltInAction {
 			return res
 		}
 
+		if min_req > 0 {
+			total = report.TotalLines
+			parsed = report.CoveredLines
+			cov_percent := float(parsed) / total * 100
```
Comment (bnjjj) [inline engine/worker/builtin_coverage.go]: it won't compile.

You haven't declared and initialise `total` and `parsed`. I suggest you to delete `total` and `parsed` and just `cov_percent := (float64(report.CoveredLines) / float64(report.TotalLines)) * 100`

File: engine/api/action/builtin.go [hunk 2]
```diff
@@ -56,6 +56,11 @@ Parse given file to extract coverage results.`
 		Description: `Path of the coverage report file.`,
 		Type:        sdk.StringParameter,
 	})
+	cover.Parameter(sdk.Parameter{
+		Name:        "minimum",
+		Description: `Minimum percentage of coverage required (-1 means no minimum).`,
+		Type:        sdk.StringParameter,
```
Comment (bnjjj) [inline engine/api/action/builtin.go]: maybe it should be an advanced parameter

File: engine/worker/builtin_coverage.go (hunk 1 above)
Comment (ziirish) [reply engine/worker/builtin_coverage.go]: Oups, sorry wrong refactoring before pushing the patch.

File: engine/api/action/builtin.go (hunk 2 above)
Comment (ziirish) [reply engine/api/action/builtin.go]: I don't know what an advanced parameter is :/
Do you have any pointer on this?

Comment (ziirish) [reply engine/api/action/builtin.go]: done

File: engine/api/action/builtin.go [hunk 3]
```diff
@@ -56,6 +56,12 @@ Parse given file to extract coverage results.`
...
```
Comment (bnjjj) [inline engine/api/action/builtin.go]:
```suggestion
		Type:        sdk.NumberParameter,
```

Comment (bnjjj) [reply engine/api/action/builtin.go]: Sorry I missed that

**Pull request merged**